### PR TITLE
[ws_pcan_js] Credit Card Disclosure

### PR DESCRIPTION
**Pull request by eakirtas**

### CODE REVIEW

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+
+// This script scan for the presence of Personally Information Identifiable in incoming WebSocket Messages.
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: "scan" > "scans"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// More Specific, it passively scan messages for credit card numbers
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: "Specific" > "specifically"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                matches.forEach(function(match){
+                    if(validateLuhnCheckSum(match)){
+
+                        helper.newAlert()
+                            .setRiskConfidence(RISK_HIGH, CONFIDENCE_HIGH)
+                            .setName("Personally Identifiable Information via WebSocket (script)")
+                            .setDescription("The response contains Personally Identifiable Information,"
+                                            + " such as CC number, SSN and similar sensitive data."
+                                            + "Credit Card type detected: " + creditCardType + ".")
+                            .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: Fixed in fd45e329c

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js (hunk 2 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: Fixed in fd45e329c

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js (hunk 3 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js]: Fixed in cc7091dbd

**Pull request closed without merging**